app/javascript/components/StyleFoundation/Box.jsx

```diff
@@ -0,0 +1,21 @@
+import styled from 'styled-components'
+import * as R from 'ramda'
+
+const toPixelIfDefined = (size) => !R.isNil(size) && `${size}px`
+
+export const Box = styled.div`
+  height: ${props => props.height};
+  width: ${props => props.width};
+`
+
+export const MarginBox = styled(Box)`
+  margin: ${props => toPixelIfDefined(props.margin)};
+  margin-left: ${props => toPixelIfDefined(props.marginL)};
+  margin-right: ${props => toPixelIfDefined(props.marginR)};
+  margin-top: ${props => toPixelIfDefined(props.marginT)};
+  margin-bottom: ${props => toPixelIfDefined(props.marginB)};
+`
+
+export const FlexBox = styled(MarginBox)`
+  display: flex;
+`
```

Comment:
These three things are components in their own right. Typically we have them as separate files just under `/components`.
Also, why not just support margin inside Box? as it stands, FlexBox doesn't support height and width.

Reply:
I understand now, thanks for the 🍐